src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/TagHelperChunkDecorator.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Chunks;
+using Microsoft.AspNetCore.Razor.CodeGenerators;
+using Microsoft.AspNetCore.Razor.CodeGenerators.Visitors;
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Host.Internal
+{
+    public class TagHelperChunkDecorator : IChunkVisitor
+    {
+        private readonly string _className; 
+        private readonly string _namespaceName;
+
+        public TagHelperChunkDecorator(CodeGeneratorContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            _namespaceName = context.RootNamespace;
+            _className = context.ClassName;
+        }
+
+        public void Accept(IList<Chunk> chunks)
+        {
+            if (chunks == null)
+            {
+                throw new ArgumentNullException(nameof(chunks));
+            }
+
+            foreach (var chunk in chunks)
+            {
+                Accept(chunk);
+            }
+        }
+
+        public void Accept(Chunk chunk)
+        {
+            if (chunk == null)
+            {
+                throw new ArgumentNullException(nameof(chunk));
+            }
+
+            var tagHelperChunk = chunk as TagHelperChunk;
```

Comment:
Can a chunk be both a `TagHelperChunk` and a `ParentChunk`? If not, we could use a `is` check here instead to make it easier to follow where this code ends up in.